### DISCUSSION TITLE
Any way to prevent jobs of the same group from running simultaneously?

### DISCUSSION THREAD

**Discussion by tarockx** [quartznet/quartznet#2070] [Q&A]
Hello

I've got a Quartz.net scheduler where I set the `quartz.threadPool.maxConcurrency` property to `5`, so I can have multiple jobs running in parallel.

This works correctly, however I have another requirement: the scheduler runs several types of jobs but I would like to **prevent jobs with the same job key / job group from running concurrently**. Is there any mechanism to achieve this?

**Comment by lahma** [comment]
I believe the FAQ has you covered: https://www.quartz-scheduler.net/documentation/faq.html#how-do-i-keep-a-job-from-firing-concurrently

**Reply by tarockx** [reply]
Thanks, but that documentation shows you how to prevent concurrent execution of jobs of the same **class**. What I would like to do is prevent concurrent execution of jobs with the same job key / job group. Is that possible?

**Reply by lahma** [reply]
Quartz should make the decision based on the job key.
